### PR TITLE
New ORT API Interface, large constant folding speedup

### PR DESCRIPTION
**Pull request by orausch**

### CODE REVIEW

File: daceml/transformation/constant_folding.py [hunk 1]
```diff
@@ -41,9 +42,6 @@
                         'ONNXTreeEnsembleRegressor'}
 # yapf: enable
 
-global UNIQUE_ID
```
Comment (tbennun) [inline daceml/transformation/constant_folding.py:44]: :)

File: doc/overviews/development.rst [hunk 2]
```diff
@@ -55,3 +55,36 @@ implementation that is used, tests can be decorated with the following two marke
...
+    import subprocess
+    import shlex
+
+    import torch
+    # initialize torch cuda context
+    a = torch.ones(1, 1).cuda()
+
+
+    def _get_gpu_mem_usage():
+        result = subprocess.check_output(shlex.split("nvidia-smi -x -q"))
```
Comment (tbennun) [inline doc/overviews/development.rst:76]: you could get the cuda free memory using the cuda driver API (`cuMemGetInfo`) but no need for that now.

**Review by tbennun — APPROVED** [review]

File: .github/workflows/gpu-ci.yml [hunk 3]
```diff
@@ -28,7 +28,8 @@ jobs:
 
       - name: Test with pytest
         env:
-          PYTEST_ARGS: --cov=daceml --cov-report=term --cov-report xml --cov-config=.coveragerc --gpu-only -m "not slow"
+          DACEML_LOG_LEVEL: debug
```
[the remark this answers is not in the document]
Comment (orausch) [reply .github/workflows/gpu-ci.yml:31]: Oops

File: daceml/ort_api/raw_api_bindings.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+from dace.codegen import codeobject, targets, compiler
+
+
+class ORTAPIError(RuntimeError):
+    """ Error thrown when an ORT function returns a non-zero Status. """
+    pass
+
+
+class keydefaultdict(collections.defaultdict):
```
[the remark this answers is not in the document]
Comment (orausch) [reply daceml/ort_api/raw_api_bindings.py:15]: This allows you to access the key that is being accessed, defaultdict doesnt

**Pull request merged**